File: src/main/java/io/github/tanguygab/conditionalactions/actions/types/commands/PlayerAction.java
package io.github.tanguygab.conditionalactions.actions.types.commands;

import io.github.tanguygab.conditionalactions.actions.types.Action;
import org.bukkit.OfflinePlayer;
import org.bukkit.entity.Player;

public class PlayerAction extends Action {

    public PlayerAction() {
        super("^(?i)(player|cmd|command):( )?");
    }

    @Override
    public String getSuggestion() {
        return "player: <command>";
    }

    @Override
    public void execute(OfflinePlayer player, String match) {
        String command = parsePlaceholders(player,match);

        if (player instanceof Player p) {
            getPlugin().sync(p, () -> p.performCommand(command));
            return;
        }
        getPlugin().sync(null, () -> getPlugin().getServer().dispatchCommand(getPlugin().getServer().getConsoleSender(),command));


    }

}

File: src/main/kotlin/io/github/tanguygab/conditionalactions/actions/types/commands/PlayerAction.kt
package io.github.tanguygab.conditionalactions.actions.types.commands

import io.github.tanguygab.conditionalactions.actions.Action
import org.bukkit.OfflinePlayer
import org.bukkit.entity.Player

class PlayerAction : Action("^(?i)(player|cmd|command):( )?") {
    override fun getSuggestion() = "player: <command>"

    override fun execute(player: OfflinePlayer?, match: String) {
        val command = parsePlaceholders(player, match)

        if (player is Player) {
            plugin.sync(player) { player.performCommand(command) }
            return
        }
        plugin.sync(null) { plugin.server.dispatchCommand(plugin.server.consoleSender, command) }
    }
}
